### PR TITLE
Add assignment equivalent functions

### PR DESCRIPTION
**Pull request by PietrH**
In this PR I've made changes to `deployments()`, `media()` and `observations()` so they can be used as assignment functions as well. This was first discussed (and work was started on) during our last coding sprint together. I'm using the same implementation as `purrr::pluck() <- `, and thus the same behaviour. 

I'm not quite sure of the examples I've added. I struggle showing off something you'd actually want to do, but isn't overly long. Just replacing `x$data$deployments` with `cars` doesn't make much sense. I avoided using `dplyr` functions in the examples.

Kept the function documentation short, feel free to make any additions. 

Also updated some of the existing functions to actually use these new assignment functions. I think I've caught all cases where assignment was used so far.

I'm using single quotes in the assignment of the assignment functions ( :supervillain: ), because purrr does... 

### Functionality
- [x] `deployments(x) <-`
- [x] `media(x) <-`
- [x] `observations(x) <-`
- [x] Update: https://github.com/inbo/camtrapdp/blob/06946dbf91189486a06257af322f3afed9433452/R/read_camtrapdp.R#L50
- [x] Update: https://github.com/inbo/camtrapdp/blob/901d35009517716435f4655381c6bce5ed396b01/R/filter_deployments.R#L58-L61
- [x] Update: https://github.com/inbo/camtrapdp/blob/901d35009517716435f4655381c6bce5ed396b01/R/filter_media.R#L62-L64
- [x] Update: https://github.com/inbo/camtrapdp/blob/901d35009517716435f4655381c6bce5ed396b01/R/filter_observations.R#L76-L78
- [x] Update: https://github.com/inbo/camtrapdp/blob/901d35009517716435f4655381c6bce5ed396b01/R/read_camtrapdp.R#L64-L70

### Tests
- [x] `deployments(x) <-`
- [x] `media(x) <-`
- [x] `observations(x) <-`
- [x] check coverage

### Documentation 
- [x] `devtools::document()`
- [x] update NEWS
- [x] Add example `deployments(x) <- value`
- [x] Add example `observations(x) <- value`
- [x] Add example `media(x) <- value`

### CODE REVIEW

**Comment by PietrH** [comment]
I need to check if the new tests are running a little slow, or if that was a fluke on my system just now

**Comment by PietrH** [comment]
On R CMD CHECK, `   checking package dependencies ...` is slow

**Comment by PietrH** [comment]
regarding the documentation, there is a small misrepresentation currently. The value assigned to `deployments()`, `media()` or `observations()` doesn't actually need to be a `tibble::tibble()`, it can be a regular base `data.frame()`

**Comment by PietrH** [comment]
> I need to check if the new tests are running a little slow, or if that was a fluke on my system just now

I suspect it's the use of `example_dataset()`

File: R/deployments.R [hunk 1]
```diff
@@ -13,3 +13,17 @@ deployments <- function(x) {
   check_camtrapdp(x)
   purrr::pluck(x, "data", "deployments")
 }
+
+#' @rdname deployments
+#' @inheritParams version
```
Comment (PietrH) [inline R/deployments.R]:
```suggestion
```

Do I need this line? I don't know why I added it in the first place?

**Comment by PietrH** [comment]
I'm getting 98.52% coverage, but `covr::package_coverage()` doesn't seem to work...

```R
covr::file_coverage(list.files("R/", full.names = TRUE, pattern = "R$"), test_files = list.files("tests/testthat/", full.names = TRUE, pattern = "R$"))
```

Does work. 

I wonder if this also affects codecov

**Comment by PietrH** [comment]
Should we bump the version number?

**Review by peterdesmet — APPROVED** [review]
Nice work! I have updated:

- Documentation now lists the two functions by name, cf. `pluck()` and indicates that the assignment function should only be used within other functions
- Examples are shortened to `head()` (a useless but at least short example), with indication that use outside function is not recommended.
- Shorten cli warning
- Use `pluck()` and `return()` over `assign_in()`: more readable and explicit
- Shorten tests
- Shorten news

Unrelated:
- Updated description of `@return` if it is `x`

**Comment by peterdesmet** [comment]
- Tibble vs data.frame: accessor will now always set tibble
- Slow tests: see #66 (thanks!)
- Bump version: will be done as part of #64

**Pull request merged**